generators/app/index.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+var path = require('path');
+var yeoman = require('yeoman-generator');
+var chalk = require('chalk');
+var yosay = require('yosay');
+var _ = require('lodash');
+
+/**
+ * Converts a string of the form ab bc cd into Ab Bc Cd
+ * @param {String} the typically hyphenless project name
+ */
+function toTitleCase(str) {
+  return _.startCase(_.toLower(str));
+}
+
+/**
+ * Converts a string of the form a-b-c into ABC
+ * @param {String} the project name
+ */
+function toClassName(str) {
+  return _.upperFirst(_.camelCase(str));
+}
+
+/**
+ * Converts a string of the form a-b-c into a-BC
+ * @param {String} the project name
+ */
+function toCssClassName(str) {
+  var hyphenlessProjectNameArray = str.split("-");
+  hyphenlessProjectNameArray.shift();
+  var namespacelessProjectName = hyphenlessProjectNameArray.join("-");
+  return str.replace(namespacelessProjectName, toClassName(namespacelessProjectName));
+}
+
+module.exports = yeoman.Base.extend({
+  constructor: function () {
+    yeoman.Base.apply(this, arguments);
+
+    this.argument('paramProjectName', {type: String, required: false});
+
+    if (this.paramProjectName === undefined) {
+      this.projectName = path.basename(process.cwd());
+    } else {
+      this.projectName = this.paramProjectName;
+    }
+  },
+
+  prompting: function () {
+    var done = this.async();
+
+    // Have Yeoman greet the user.
+    this.log(yosay(
+      'Welcome to the ' + chalk.red('generator-terra-module') + ' generator! This will scaffold a skeleton to help start developing a terra module!'
+    ));
+
+    var prompts = [{
+      type: 'input',
+      name: 'projectName',
+      message: 'Your module name:',
+      default: this.projectName
+    }];
+
+    this.prompt(prompts, function (props) {
+      this.props = props;
+      // To access props later use this.props.name;
+
+      done();
+    }.bind(this));
+  },
+
+  writing: function () {
+    this.fs.copyTpl(
+      this.templatePath('src/projectName.jsx'),
+      this.destinationPath('src/' + toClassName(this.props.projectName) + '.jsx'),
+      {
+        projectName: this.props.projectName,
+        projectClassName: toClassName(this.props.projectName),
```

Comment:
In the react componet jsx file, the [component gets named as `TerraModule`](https://github.com/ryanthemanuel/terra-module/blob/master/src/TerraModule.jsx#L6). I think we should prefer unprefixed component names for react components. Since ES2015 imports offer a way to alias an export, we are less likely to run into naming collisions and it is easier to resolve compared to naming collisions in CSS.

If a prefix needs to be added to avoid naming collisions, it can be done when the react component is imported.

For example, given: 
* the project name would be `terra-button`
* react component name would be `Button`
* `.jsx` file would be `TerraButton.jsx`. 

If consumer is already using a `<Button />` component, they would alias the terra `Button` component when they import it: `import {Button as TerraButton} from './TerraButton';`